src/storage/irc_transaction.cpp

```diff
@@ -1,34 +1,225 @@
+		} else if (update.has_set_snapshot_ref_update) {
+			auto update_json = yyjson_mut_arr_add_obj(doc, updates_array);
+			auto &ref_update = update.set_snapshot_ref_update;
+
+			//! updates[...].action
+			yyjson_mut_obj_add_strcpy(doc, update_json, "action", "set-snapshot-ref");
+			//! updates[...].ref-name
+			yyjson_mut_obj_add_strcpy(doc, update_json, "ref-name", ref_update.ref_name.c_str());
+			//! updates[...].type
+			yyjson_mut_obj_add_strcpy(doc, update_json, "type", ref_update.snapshot_reference.type.c_str());
+			//! updates[...].snapshot-id
+			yyjson_mut_obj_add_uint(doc, update_json, "snapshot-id", ref_update.snapshot_reference.snapshot_id);
+		} else {
+			throw NotImplementedException("Can't serialize this TableUpdate type to JSON");
+		}
+	}
+
+	//! identifier
+	D_ASSERT(table.has_identifier);
+	auto &_namespace = table.identifier._namespace.value;
+	auto identifier_json = yyjson_mut_obj_add_obj(doc, root_object, "identifier");
+
+	//! identifier.name
+	yyjson_mut_obj_add_strcpy(doc, identifier_json, "name", table.identifier.name.c_str());
+	//! identifier.namespace
+	auto namespace_arr = yyjson_mut_obj_add_arr(doc, identifier_json, "namespace");
+	D_ASSERT(_namespace.size() == 1);
+	yyjson_mut_arr_add_strcpy(doc, namespace_arr, _namespace[0].c_str());
+}
+
+void CommitTransactionToJSON(yyjson_mut_doc *doc, yyjson_mut_val *root_object,
+                             const rest_api_objects::CommitTransactionRequest &req) {
+	auto table_changes_array = yyjson_mut_obj_add_arr(doc, root_object, "table-changes");
+	for (auto &table : req.table_changes) {
+		auto table_obj = yyjson_mut_arr_add_obj(doc, table_changes_array);
+		CommitTableToJSON(doc, table_obj, table);
+	}
+}
+
+string JsonDocToString(std::unique_ptr<yyjson_mut_doc, YyjsonDocDeleter> doc) {
+	auto root_object = yyjson_mut_doc_get_root(doc.get());
+
+	//! Write the result to a string
+	auto data = yyjson_mut_val_write_opts(root_object, YYJSON_WRITE_ALLOW_INF_AND_NAN, nullptr, nullptr, nullptr);
+	if (!data) {
+		throw InvalidInputException("Could not create a JSON representation of the table schema, yyjson failed");
+	}
+	auto res = string(data);
+	free(data);
+	return res;
+}
+
+static rest_api_objects::TableRequirement CreateAssertRefSnapshotIdRequirement(IcebergSnapshot &old_snapshot) {
+	rest_api_objects::TableRequirement req;
+	req.has_assert_ref_snapshot_id = true;
+
+	auto &res = req.assert_ref_snapshot_id;
+	res.ref = "main";
+	res.snapshot_id = old_snapshot.snapshot_id;
+	res.type.value = "assert-ref-snapshot-id";
+	return req;
+}
+
 void IRCTransaction::Commit() {
-	if (transaction_state == IRCTransactionState::TRANSACTION_STARTED) {
-		transaction_state = IRCTransactionState::TRANSACTION_FINISHED;
-		//		connection.Execute("COMMIT");
+	if (dirty_tables.empty()) {
+		return;
+	}
+
+	try {
+		Connection temp_con(db);
+		//! This automatically starts a transaction
+		temp_con.BeginTransaction();
+		auto &context = temp_con.context;
```

Comment:
For my understanding, do we need temp_con just for the ClientContext? Or is there another reason for it?

Reply:
Just for the client context that is properly associated with a transaction, which is a requirement that the ClientContext in the Transaction that we are committing does not satisfy.